### ISSUE TITLE
1.7.0 build fails: error: address of array 'this->storage->audio_otherscene' will always evaluate to 'true'

### ISSUE THREAD

**Issue by yurivict** [surge-synthesizer/surge#2384]
```
/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.7.0/src/common/dsp/Oscillator.cpp:802:85: error: address of array 'this->storage->audio_otherscene' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
   if( isInSceneB && localcopy[oscdata->p[4].param_id_in_scene].f > 0.f && storage->audio_otherscene )
                                                                        ~~ ~~~~~~~~~^~~~~~~~~~~~~~~~
1 error generated.
```

clang-9
OS: FreeBSD 12.1

**Comment by baconpaul** [comment]
Sigh. Sorry we don't compile linux clang regularly. I had seen that warning go by on my mac but forgot about it. Let me push a fix.

Is there a CMake variable I can refer to that will let me know I'm on FreeBSD by the way?

Thanks

**Comment by baconpaul** [comment]
This should be fixed as of c8271d1

I don't have FreeBSD to test it; but I did activate this warning-as-error on my mac (running clang-11), got an error, and resolved the error; the commit both resolves the error and keeps it at error level on the mac.

Please do let us know if you get other issues. We are fixing a collection of small things with 1.7.0 and will probably do a 1.7.1 release soon enough including this fix and a few others we found after broad release.

**Comment by yurivict** [comment]
@baconpaul 

Is there a CMake variable I can refer to that will let me know I'm on FreeBSD by the way?

```
if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD|Linux")
endif()
```
